lightningd/peer_control.c

```diff
@@ -901,7 +901,7 @@ static struct command_result *json_close(struct command *cmd,
 {
 	const jsmntok_t *idtok;
 	struct peer *peer;
-	struct channel *channel;
+	struct channel *channel = NULL;
```

Comment:
Not sure what compiler you're using, but does putting "assert(channel);" after:
```
		res = command_find_channel(cmd, buffer, idtok, &channel);
		if (res)
			return res;
```

also suppress the warning?  That seems clearer to me...